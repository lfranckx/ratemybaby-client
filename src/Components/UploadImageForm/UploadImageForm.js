import React, { Component } from 'react'
import BabyContext from '../../Contexts/BabyContext'
import BabyApiService from '../../Services/baby-api-service'


export default class UploadImage extends Component {
    static defaultProps = {
        onLoginSuccess: () => {}
    }

    static contextType = BabyContext

    constructor(props) {
        super(props)
        this.fileInput = React.createRef()
        
        this.state = {
            error: null,
            fileSelected: null
        }
    }

    componentDidMount() {
        this.context.clearError()
    }

    handleSingleFileUpload = ev => {
        ev.preventDefault()
        this.setState({ error: null })
        const { baby } = this.context
        const  fileSelected  = this.fileInput.current.files[0]
        const data = new FormData()
        data.append('image', fileSelected)
        
        BabyApiService.postImageFile(data)
        .then(res => {
            (!res.ok)
                ? res.json().then(e => Promise.reject(e))
                : res.json()
                .then(data => {
                    baby.image_url = data.image_url
                    BabyApiService.updateBaby(baby)
                    this.props.onUploadSuccess()
                })
        })
    }

    render() {
        const { error } = this.state

        return (
                <form 
                    id="upload-form"
                    onSubmit={this.handleSingleFileUpload}
                >
                    <div role='alert'>{error && <p className='error'>{error}</p>}</div>
                    <label>Select an image to upload:</label>
                    <input 
                        ref={this.fileInput}
                        type="file"
                        accept=".png, .jpg, .jpeg .gif"
                        name="file" 
                        id="file" />
                    <button id="upload-button" type="submit">Submit</button>
                </form>
        )
    }
}
