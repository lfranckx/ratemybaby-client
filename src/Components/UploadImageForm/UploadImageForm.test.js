import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import UploadImageForm from './UploadImageForm';

const simulateChangeOnInput = (wrapper, inputSelector, newValue) => {
    const input = wrapper.find(inputSelector);
    input.simulate('change', {
        target: {value: newValue}
    });
    return wrapper.find(inputSelector);
};

describe(`UploadImageForm component`, () => {
    const props = {
        babies: {},
        handleUploadSuccess: () => {}
    }

    it(`renders a form given props`, () => {
        const wrapper = shallow(<UploadImageForm {...props}/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    });

    it(`lets me select a file`, () => {
        const wrapper = shallow(<UploadImageForm {...props}/>);
        simulateChangeOnInput(wrapper, '#file', 'file.jpeg');
    });
})