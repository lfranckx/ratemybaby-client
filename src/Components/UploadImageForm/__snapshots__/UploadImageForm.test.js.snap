// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadImageForm component renders a form given props 1`] = `
<form
  id="upload-form"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <label>
    Select an image to upload:
  </label>
  <input
    accept=".png, .jpg, .jpeg .gif"
    aria-label="file"
    id="file"
    name="file"
    type="file"
  />
  <button
    id="upload-button"
    type="submit"
  >
    Submit
  </button>
</form>
`;
